### DISCUSSION TITLE
Problem when mapping long corridor

### DISCUSSION THREAD

**Discussion by MartiBolet** [stella-cv/stella_vslam#189] [Q&A] [answered]
Hello,

One problem I'm facing right now is that when i'm mapping using a fisheye camera, when I'm moving in a straight line, the path of the keyframes is getting a curvature all the way long. If I'm moving towards a long corridor, at the start and at the end, the curvature makes an important prounciation. It should be noted taht the camera is always at the same height and only moves parallel to the ground.

As an example, the following image contains a square zone, plus the corridor in both directons, and as the image shows, the path of the keyframes are making a curvature:

![MappingLongCorridor](https://user-images.githubusercontent.com/43337758/141103831-8b351d00-b348-494e-b1ae-f21bcfa799bb.png)

As I found this problem, I assumed that the problem comes from the calibration, and I tried to remake the calibration of the fisheye many times, getting the same error. So right now I don't have any clue, anyone knows if this is a calibration problem or something else? If it's a calibration problem, which is the best program/practices to make a better calibration? Any help is appreciated.

Thanks in advance,
Martí Bolet Boixeda

**Comment by ymd-stella** [comment, marked as answer]
Are you pointing your camera at the ceiling? I suspect that is the cause.

**Reply by MartiBolet** [reply]
Yes, the camera is facing the ceiling. This has some implication at the code or what?

Thank you,

**Reply by ymd-stella** [reply]
Long-range keyframe matching becomes difficult, so accuracy is degraded.

**Comment by technick100** [comment]
Hello. Could you please say how to calibrate camera?

**Reply by 2black0** [reply]
https://docs.opencv.org/4.x/dc/dbb/tutorial_py_calibration.html

the calibration process is quite easy to do with opencv
